Bill3.java

```diff
@@ -1,8 +1,15 @@
 // Splits a restaurant bill evenly among three diners.
+
+
 public class Bill3 {
 	public static void main(String[] args) {
 		// To get you started, here is the first line in the program:
 	    String name1 = args[0];
-	    // Replace this comment with the rest of your code   
+		String name2 = args[1];
+		String name3 = args[2];
+	    int bill = Integer.parseInt(args[3]);
+		
+		System.out.println("Dear " +name3+", "+name2+ ", and "+name1+": pay " +Math.ceil(((double)bill)/3)+" Shekels each.");
```

Comment:
- The math formula for calculating the bill is correct, but writing it inside the print function is less readable. 
Optional - define it before the println as a variable: 
splittedBill = Math.ceil(((double)bill)/3)

- Another thing to keep in mind - if you want the bill to be a double, parse it as a double from the beginning - 
**double** bill = **Double**.parseDouble